packages/slider/index.tsx

```diff
@@ -0,0 +1,677 @@
+import {EventType as K, SpecificEventListener} from '@material/base/types';
+import {MDCSliderAdapter, MDCSliderFoundation} from '@material/slider';
+import {cssClasses, numbers} from '@material/slider/constants';
+import classnames from 'classnames';
+import React from 'react';
+
+const KEY_IDS = {
+  ARROW_DOWN: 'ArrowDown',
+  ARROW_LEFT: 'ArrowLeft',
+  ARROW_RIGHT: 'ArrowRight',
+  ARROW_UP: 'ArrowUp',
+  END: 'End',
+  HOME: 'Home',
+  PAGE_DOWN: 'PageDown',
+  PAGE_UP: 'PageUp',
+};
+
+type UpEventType = 'mouseup' | 'pointerup' | 'touchend';
+type DownEventType = 'mousedown' | 'pointerdown' | 'touchstart';
+type MoveEventType = 'mousemove' | 'pointermove' | 'touchmove';
+type MouseLikeEvent = MouseEvent | PointerEvent | TouchEvent;
+type ReactMouseLikeEvent =
+  | React.MouseEvent
+  | React.PointerEvent
+  | React.TouchEvent;
+
+type MoveEventMap = {readonly [K in DownEventType]: MoveEventType};
+
+const UP_EVENTS: UpEventType[] = ['mouseup', 'pointerup', 'touchend'];
+
+const MOVE_EVENT_MAP: MoveEventMap = {
+  mousedown: 'mousemove',
+  pointerdown: 'pointermove',
+  touchstart: 'touchmove',
+};
+
+export interface SliderProps extends React.HTMLProps<HTMLDivElement> {
+  value: number;
+  discrete?: boolean;
+  displayMarkers?: boolean;
+  min: number;
+  max: number;
+  step?: number;
+  disabled?: boolean;
+  className?: string;
+  tabIndex?: number;
+  isRtl?: boolean;
+  onValueChange?: (value: number) => void;
+  onValueInput?: (value: number) => void;
+}
+
+interface SliderState {
+  classList: Set<string>;
+  markerValue?: number;
+  markerStyles: React.CSSProperties[];
+}
+
+class Slider extends React.Component<SliderProps, SliderState> {
+  get classes() {
+    const {classList} = this.state;
+    const {className, discrete, displayMarkers} = this.props;
+    return classnames('mdc-slider', Array.from(classList), className, {
+      'mdc-slider--discrete': discrete,
+      'mdc-slider--display-markers': discrete && displayMarkers,
+    });
+  }
+
+  get adapter(): MDCSliderAdapter {
+    return {
+      /**
```

Comment:
can you remove all the adapter comments? I appreciate the thoroughness however it makes the adapter 200 lines.